action.yml

```diff
@@ -20,18 +20,21 @@ outputs:
 runs:
   using: 'composite'
   steps:
-    - uses: actions/setup-python@13ae5bb136fac2878aff31522b9efb785519f984  # v4.3.0
+    - run: echo "${{ runner.name }}"
+      shell: bash
+    - uses: thebrowsercompany/check-python-exists@462c75d86b49a66ae75053cbc2f6a43e8f7a84b8 # PR update when merged
       id: setup-python
       with:
         python-version: ${{ inputs.python-version }}
+        is-self-hosted: ${{ !startsWith('GitHub Actions', runner.name) }}
```

Comment:
cc @maruel this is what I mean by being able to infer within the caller. It isn't perfect though because it does not offer a way to override in this action. But not sure if important really.